modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/mapping/MappingServiceImpl.java

```diff
@@ -85,6 +94,10 @@ public MappingServiceImpl(
         this.targetProvider = targetProvider;
         this.templatesCache = cacheFactory.create(cacheSize);
         this.taskExecutor = taskExecutor;
+        this.mappingsCache = Caffeine.newBuilder()
```

Comment:
Added two additional methods to internal cache interface.
`compute`
`removeIfValue` (this is debatable - instead of this weird method we can add a `values()` method). :thinking: